src/main/webapp/ratingScript.js

```diff
@@ -0,0 +1,128 @@
+    translate: document.getElementById("translate").value,
+  };
+  document.getElementById("term-form").reset();
+  const url = addingAttributesToURL(ratingProperties);
+  return [url, ratingProperties];
+}
+
+function addingAttributesToURL(ratingProperties) {
+  const url = new URL("/data", window.location.origin);
+  url.searchParams.set("course-key", ratingProperties.courseKey);
+  url.searchParams.set("term-key", ratingProperties.termKey);
+  url.searchParams.set("hour", ratingProperties.hours);
+  url.searchParams.set("difficulty", ratingProperties.difficulty);
+  url.searchParams.set("term-input", ratingProperties.termInput);
+  url.searchParams.set("prof-input", ratingProperties.profInput);
+  url.searchParams.set("rating-term", ratingProperties.ratingTerm);
+  url.searchParams.set("rating-prof", ratingProperties.ratingProf);
+
+  return url;
+}
+
+async function passRatingProperties() {
+  document.getElementById("message-rating").innerHTML = "";
+  const urlAndData = await getRatingPropertiesToStore();
+  postRatingProperties(urlAndData[0], urlAndData[1]);
+}
+
+$(function () {
+  $('[data-toggle="tooltip"]').tooltip();
+});
+
+async function getLatestRating() {
+  document
+    .getElementById("message-rating")
+    .classList.remove("alert-light");
+  document
+    .getElementById("message-rating")
+    .classList.add("alert-secondary");
+  document.getElementById("message-rating").innerHTML =
+    "Fetching your rating...";
+
+  const userId = await verify();
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const termKey = urlParams.get("term-key");
+
+  const url = new URL("/latest-rating", window.location.origin);
+  url.searchParams.set("reviewer-id", userId);
+  url.searchParams.set("term-key", termKey);
+  const response = await fetch(url);
+  const formInfo = await response.json();
+
+  if (Object.keys(formInfo).length == 0) {
+    document
+      .getElementById("message-rating")
+      .classList.remove("alert-secondary");
+    document
+      .getElementById("message-rating")
+      .classList.add("alert-danger");
+    document.getElementById("message-rating").innerHTML =
+      "You have not submitted a rating for this term";
+  } else {
+    document.getElementById("term-input").innerHTML = formInfo["termInput"];
+    document.getElementById("prof-input").innerHTML = formInfo["profInput"];
+    document.getElementById("rating-term").value = formInfo["ratingTerm"];
+    document.getElementById("rating-prof").value = formInfo["ratingProf"];
+    document.getElementById("hours").value = formInfo["hours"];
+    document.getElementById("difficulty").value = formInfo["difficulty"];
+    document.getElementById("grade").value = formInfo["grade"];
+    document.getElementById("translate").value = formInfo["translation"];
+
+    document.getElementById("message-rating").innerHTML =
+      "Your form has been populated!";
+    document
+      .getElementById("message-rating")
```

Comment:
FYI everytime you do this you force the DOM to layout itself, then when you modify the class list it has to reflow. 
This is called layout thrashing.

Idelaly, declare a variable up top with messageRatingElement inside, then use that everywhere. It's also a bit cleaner code.

Reply:
Ohh did not know that, changing that now!